backend/fms_core/services/experiment_run_info.py

```diff
@@ -185,7 +181,15 @@ def _generate_pooled_samples(experiment_run: ExperimentRun, pool: Sample) -> Lis
         derived_by_sample: DerivedBySample = DerivedBySample.objects.get(derived_sample=derived_sample, sample=pool)
         run_info_sample.pool_volume_ratio = float(derived_by_sample.volume_ratio)
 
-        run_info_samples.append(run_info_sample)
+        project: Optional[Project] = derived_by_sample.project
+        if project is None:
+            raise Exception(f'Sample {derived_by_sample.sample} has no project.')
+        else:
+            run_info_sample.project_obj_id = project.id
+            run_info_sample.project_name = project.name
+            run_info_sample.external_project_id = project.external_id
+            run_info_sample.external_project_name = project.external_name
+            run_info_samples.append(run_info_sample)
```

Comment:
you actually adding type isn't convincing my editor that it is indeed an optional project ;)

Reply:
but do still keep it

Reply:
those are hints not types ... they do not have any functional meaning ... your editor is right to remain doubtful.